tests/cli/test_doctor.py

```
@@ -253,9 +253,13 @@ def test_detect_pipx_new_share_location(self):
         """Should detect pipx when executable is under PIPX_HOME/venvs/."""
+        import os
 
-        custom_home = "/custom/pipx"
-        fake_exe = "/custom/pipx/venvs/fo-core/bin/python"
+        # Build paths with the OS-native separator so the test matches what
+        # _detect_install_method() constructs via os.path.join + os.sep
+        # (forward slashes on POSIX, backslashes on Windows).
+        custom_home = os.path.join(os.sep, "custom", "pipx")
+        fake_exe = os.path.join(custom_home, "venvs", "fo-core", "bin", "python")
         with patch("cli.doctor.sys.executable", fake_exe):
             with patch.dict("os.environ", {"PIPX_HOME": custom_home}):
```

Ruff (0.15.15)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)